### ISSUE TITLE
Conflicting cross-version suffixes with SBT 1.9.0-RC2

### ISSUE THREAD

**Issue by mdedetrich** [sbt/sbt#7243]
## steps

After https://github.com/sbt/sbt/issues/7233 was resolved with the new release of SBT 1.9.0-RC2, I updated the noted PR with latest version of SBT and while it did solve that specific problem, there now appears to be a new one.

## problem

For builds using cross-version that also happen to produce sbt-plugins, SBT 1.9.0-RC2 produces the following exception

```
[info] [error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:/tmp/sbt_d4495aa5/"), "sbt_d4495aa5"):
[info] [error]    org.scala-lang.modules:scala-collection-compat _3, _2.13
[info] [error]    com.thesamet.scalapb:lenses _3, _2.13
[info] [error]    com.thesamet.scalapb:scalapb-runtime _3, _2.13
[info] [error] java.lang.RuntimeException: Conflicting cross-version suffixes in: org.scala-lang.modules:scala-collection-compat, com.thesamet.scalapb:lenses, com.thesamet.scalapb:scalapb-runtime
```

see https://github.com/apache/incubator-pekko-grpc/actions/runs/4920031532/jobs/8788388321?pr=64 for more information.

## expectation

That upgrading SBT to 1.9.0-RC2 works without any regressions.

## notes

Similar to the previously noted PR my hunch is that its due to https://github.com/sbt/sbt/pull/7096 since RC2 didn't add any new relevant features compared to RC1, hence I am pinging @adpi2 .

**Comment by adpi2** [comment]
This does not seem to be a bug but an intended error report that was added in sbt 1.5.5 by https://github.com/sbt/librarymanagement/pull/383. It means that you have a conflict in your dependency graph between two similar dependencies with different binary versions.

You have a direct dependency to `scalapb-runtime_3`, but also to `pekko-grpc-runtime_2.13` that depends on `scalapb-runtime_2.13`.

In https://github.com/apache/incubator-pekko-grpc/pull/64, in the scripted `scala3/01-basic-client-server`, if I change the sbt version from 1.9.0-RC2 back to 1.5.5, I got the same error report. The error disappears on sbt 1.5.4.
